container/dev/docker-compose.yml

```diff
@@ -0,0 +1,24 @@
+version: '3'
+
+services:
+
+  postgres:
+    image: postgres
+    volumes:
+      - .pgdata:/var/lib/postgresql/data
+    env_file:
+      - ../postgres.env
+
+  web:
+    build:
+      context: ../..
+      dockerfile: container/dev/Dockerfile
+    command: sh -c "pip install -e . && bounce start"
```

Comment:
Doesn't `pip install` already occur in your `Dockerfile`? It's probably better to let installs and stuff all happen during the build, and do straight into `bounce start`, which can be:

```
# container/dev/Dockerfile
ENTRYPOINT ["bounce", "start"]
```

Bringing this up because in the early days when I was testing Sleuth with Inertia, having dependency installations during `docker-compose up` via `command` caused all sorts of problems (namely that it's easier to catch install problems during build, while during `docker-compose up` a problem during `command` just fails silently, which is problematic because `docker-compose` doesn't do very good reporting on when only 1 out of 3 containers go down)

Reply:
You raise a good point but there's a good reason I left this out of the Dockerfile. This `pip install` is different to the pip install we do in the Dockerfile because it's installing _our_ code as an "editable" Python package. 

Here's what would happen if I put `pip install -e .` and then used `bounce` as the entrypoint in the Dockerfile:

1. Docker is building the image, and it has just added the project directory to the image (`ADD . /opt/bounce`)
2. Docker executes `RUN pip install -e .` and installs our code as a Python package in editable mode
3. Docker-compose runs the container built from this image and mounts the project directory to `/opt/bounce` *over* the directory that we used to install the package in editable mode
4. An `DistributionNotFound` raised when Docker runs `bounce start` because the directory we originally used to install the `bounce` package has been overridden by the mount in our `docker-compose`

I can't get rid of the mount either, because the whole point of running a `dev` container is that we can edit code and immediately run it without having to re-create the image/container.

Reply:
> You raise a good point but there's a good reason I left this out of the Dockerfile. This pip install is different to the pip install we do in the Dockerfile because it's installing our code as an "editable" Python package.

Ahhh I see, that's pretty neat, makes sense to do it this way then - thanks for clearing that up!

> all you have to do is docker-compose logs web to see what happened, but you'll see the error in the logs straight away if your just running docker-compose up using this compose file

This is alright in normal use cases where someone is actively paying attention, but I have yet to find a good way to automate this in situations like Inertia deployments 🤔

Reply:
Yea unfortunately docker-compose doesn't quite give you all the control you need here. The only alternative I can see is having script that basically does what docker-compose does:

```bash
docker build ...
docker network create ...
docker run --network <network> <other args> <web image>
docker run --network <network> <other args> <postgres image>
```